frontend/src/metabase/lib/cron.ts

```diff
@@ -0,0 +1,64 @@
+import { t } from "ttag";
+import { isValidCronExpression } from "cron-expression-validator";
+import cronstrue from "cronstrue";
+
+import { has24HourModeSetting } from "metabase/lib/time";
+import MetabaseSettings from "metabase/lib/settings";
+
+function translateErrorMessage(message: string) {
+  const errorMessageMap: Record<string, string> = {
+    "Day-of-Week values must be between 1 and 7": t`Day-of-week values must be between 1 and 7`,
+    "Day-of-Week values must be SUN, MON, TUE, WED, THU, FRI, SAT OR between 1 and 7, - * ? / L #": t`Day-of-week values must be SUN, MON, TUE, WED, THU, FRI, SAT OR between 1 and 7, - * ? / L #`,
+    "(Day of week) - Unsupported value for field. Possible values are 1-7 or SUN-SAT , - * ? / L #": t`Unsupported day of week value. Possible values are 1-7 or SUN-SAT , - * ? / L #`,
+    "A numeric value between 1 and 5 must follow the # option": t`A numeric value between 1 and 5 must follow the # option`,
+    "Day of month values must be between 1 and 31": t`Day of month values must be between 1 and 31`,
+    "Offset from last day must be <= 30": t`Offset from last day must be less or equal 30`,
+    "Month values must be between 1 and 12": t`Month values must be between 1 and 12`,
+    "Month values must be JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC OR between 1 and 12": t`Month values must be JAN, FEB, MAR, APR, MAY, JUN, JUL, AUG, SEP, OCT, NOV, DEC or between 1 and 12`,
+    "Start year must be less than stop year": t`Start year must be less than stop year`,
+    "(Year) - Unsupported value for field. Possible values are 1970-2099 , - * /": t`Unsupported year value. Possible values are 1970-2099 , - * /`,
+    "Minute and Second values must be between 0 and 59 and Hour Values must be between 0 and 23": t`Minute and second values must be between 0 and 59 and hour values must be between 0 and 23`,
+    "? can only be specfied for Day-of-Month -OR- Day-of-Week": t`? can only be specified for day-of-month or day-of-week`,
```

Comment:
This `?` is a bit tricky and this error message isn't very clear. I suggest:

`You must use ? in the day-of-week or day-of-month field`